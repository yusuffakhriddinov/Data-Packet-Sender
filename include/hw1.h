/*
ANY CHANGES MADE TO THIS FILE WILL BE DISCARDED DURING GRADING!
*/

#ifndef HW1_H
#define HW1_H

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void print_packet_sf(unsigned char packet[]);
unsigned int compute_checksum_sf(unsigned char packet[]);
unsigned int reconstruct_array_sf(unsigned char *packets[], unsigned int packets_len, int *array, unsigned int array_len);
unsigned int packetize_array_sf(int *array, unsigned int array_len, unsigned char *packets[], unsigned int packets_len,
                          unsigned int max_payload, unsigned int src_addr, unsigned int dest_addr, 
                          unsigned int src_port, unsigned int dest_port, unsigned int maximum_hop_count,
                          unsigned int compression_scheme, unsigned int traffic_class);

#endif // HW1_H
