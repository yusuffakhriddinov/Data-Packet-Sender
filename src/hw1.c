#include "hw1.h"

void print_packet_sf(unsigned char packet[])
{
    (void)packet;
}

unsigned int compute_checksum_sf(unsigned char packet[])
{
    (void)packet;
    return -1;
}

unsigned int reconstruct_array_sf(unsigned char *packets[], unsigned int packets_len, int *array, unsigned int array_len) {
    (void)packets;
    (void)packets_len;
    (void)array;
    (void)array_len;
    return -1;
}

unsigned int packetize_array_sf(int *array, unsigned int array_len, unsigned char *packets[], unsigned int packets_len,
                          unsigned int max_payload, unsigned int src_addr, unsigned int dest_addr,
                          unsigned int src_port, unsigned int dest_port, unsigned int maximum_hop_count,
                          unsigned int compression_scheme, unsigned int traffic_class)
{
    (void)array;
    (void)array_len;
    (void)packets;
    (void)packets_len;
    (void)max_payload;
    (void)src_addr;
    (void)dest_addr;
    (void)src_port;
    (void)dest_port;
    (void)maximum_hop_count;
    (void)compression_scheme;
    (void)traffic_class;
    return -1;
}
