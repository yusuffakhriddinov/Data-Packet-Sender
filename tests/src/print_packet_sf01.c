#include <stdio.h>
#include "hw1.h"

int main() {
    unsigned char packet[] = {
			0x01,0xd2,0x08,0xa0,0xb4,0x11,0xaa,0xcd,0x00,0x00,0x01,0xca,0xde,0xad,0xb1,0xf3,0x00,0x84,0x5f,0xed,0xff,0xff,0x66,0x8f,0x05,0x88,0x81,0x92,
		};
    print_packet_sf(packet);
    return 0;
}
