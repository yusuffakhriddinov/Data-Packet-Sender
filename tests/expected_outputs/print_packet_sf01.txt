Source Address: 1908874
Destination Address: 11801002
Source Port: 12
Destination Port: 13
Fragment Offset: 0
Packet Length: 28
Maximum Hop Count: 21
Checksum: 6204849
Compression Scheme: 3
Traffic Class: 51
Payload: 8675309 -39281 92832146
